reports-backend/src/main/java/org/jboss/da/reports/api/AdvancedArtifactReport.java

```diff
@@ -0,0 +1,117 @@
+                modulesAnalyzed.add(dep.getGav());
+                populateAdvancedFields(dep, modulesAnalyzed);
+            } else {
+                // only generate populate advanced report with community GAVs
+                if (VersionParser.isRedhatVersion(dep.getVersion())) continue;
+
+                // we have a top-level module dependency
+                if (dep.isWhitelisted()) whitelistArtifacts.add(dep.getGav());
+                if (dep.isBlacklisted()) blacklistArtifacts.add(dep.getGav());
+
+                if (dep.getBestMatchVersion().isPresent()) {
+                    communityGavsWithBestMatchVersions.add(dep.getGav());
+                    // also add bestMatchVersion GAV to set
+                    communityGavsWithBestMatchVersions.add(new GAV(dep.getGroupId(),
+                            dep.getArtifactId(), dep.getBestMatchVersion().get()));
+                } else if (!dep.getAvailableVersions().isEmpty()) {
+                    communityGavsWithBuiltVersions.add(dep.getGav());
+
+                    // also add available versions to set
+                    dep.getAvailableVersions().stream()
+                            .forEach(v -> communityGavsWithBuiltVersions
+                                    .add(new GAV(dep.getGroupId(), dep.getArtifactId(), v)));
+
+                    // add any whitelisted GAVs, if any
+                    addWhitelistedGAVToCommunityGavsWithBuiltVersions(dep.getGroupId(), dep.getArtifactId());
+                } else if (hasWhitelistedGA(dep.getGroupId(), dep.getArtifactId())) {
+                    // no bestmatch version, no built versions, but we have GAVs whitelisted
+                    addWhitelistedGAVToCommunityGavsWithBuiltVersions(dep.getGroupId(), dep.getArtifactId());
+                } else {
+                    communityGavs.add(dep.getGav());
+                }
+            }
+        }
+    }
+
+    private boolean isDependencyAModule(ArtifactReport dependency) {
+        return dependency.getGroupId().equals(artifactReport.getGroupId())
+                && dependency.getVersion().equals(artifactReport.getVersion());
+    }
+
+    private boolean hasWhitelistedGA(String groupId, String artifactId) {
+        if (whiteArtifactService.getAll() == null) {
+            return false;
+        }
+        return whiteArtifactService.getAll().stream()
+                .anyMatch(white -> white.getArtifactId().equals(artifactId) && white.getGroupId().equals(groupId));
+    }
+
+    private void addWhitelistedGAVToCommunityGavsWithBuiltVersions(String groupId, String artifactId) {
+        if (whiteArtifactService.getAll() == null) {
+            return;
+        }
+        whiteArtifactService.getAll().stream()
+                .filter(white -> white.getArtifactId().equals(artifactId) && white.getGroupId().equals(groupId))
+                .forEach(white -> communityGavsWithBuiltVersions.add(new GAV(groupId, artifactId, white.getVersion())));
```

Comment:
I don't thik that every built version is in whitelist. We should use VersionFinder.